Source/Orts.Simulation/Simulation/RollingStocks/MSTSSteamLocomotive.cs

```diff
@@ -5447,6 +5453,11 @@ private void UpdateFiring(float absSpeedMpS)
                     Injector2IsOn = false;
                     Injector2Fraction = 0.0f;
                     InjectorLockedOut = true;
+                    if (!Injector1SoundIsOn) // turn injector sound on only when injector starts
+                    {
+                        Injector1SoundIsOn = true;
+                        SignalEvent(Event.WaterInjector1On);
+                    }
```

Comment:
I notice you repeat this logic many times. Consider factoring it out into a helper method?

Reply:
Unfortunately I am a "self taught" coder, so always happy to have examples of better ways to code stuff and achieve the same outcome.

Thanks

Reply:
See 7952b1944047a2197044f4b9707af1579a0431e5.